libr/io/p/io_dsc.c

```diff
@@ -255,119 +255,194 @@ static ut64 __lseek_dsc(RIO *io, RIODesc *fd, ut64 offset, int whence) {
+							if (pj) {
 								pj_kn (pj, "diversity", diversity);
+							} else if (sb) {
+								r_strbuf_appendf (sb, "diversity: 0x%"PFMT64x"\n", diversity);
 							}
-							ut64 key = (raw_value >> 49) & 3;
-							const char * names[4] = { "ia", "ib", "da", "db" };
+						}
+						ut64 key = (raw_value >> 49) & 3;
+						const char * names[4] = { "ia", "ib", "da", "db" };
+						if (pj) {
 							pj_ks (pj, "key", names[key]);
+						} else if (sb) {
+							r_strbuf_appendf (sb, "key: %s\n", names[key]);
 						}
-						break;
-					case 5:
-						if (R_IS_PTR_AUTHENTICATED (raw_value)) {
-							bool has_diversity = (raw_value & (1ULL << 50)) != 0;
+					}
+					break;
+				case 5:
+					if (R_IS_PTR_AUTHENTICATED (raw_value)) {
+						bool has_diversity = (raw_value & (1ULL << 50)) != 0;
+						if (pj) {
 							pj_kb (pj, "has_diversity", has_diversity);
-							if (has_diversity) {
-								ut64 diversity = (raw_value >> 34) & 0xFFFF;
+						}
+						if (has_diversity) {
+							ut64 diversity = (raw_value >> 34) & 0xFFFF;
+							if (pj) {
 								pj_kn (pj, "diversity", diversity);
+							} else if (sb) {
+								r_strbuf_appendf (sb, "diversity: 0x%"PFMT64x"\n", diversity);
 							}
-							ut64 key = (raw_value >> 51) & 1;
-							const char * names[2] = { "ia", "da" };
+						}
+						ut64 key = (raw_value >> 51) & 1;
+						const char * names[2] = { "ia", "da" };
+						if (pj) {
 							pj_ks (pj, "key", names[key]);
+						} else if (sb) {
+							r_strbuf_appendf (sb, "key: %s\n", names[key]);
 						}
-						break;
-					default:
-						R_LOG_ERROR ("Unsupported rebase info version %d", trimmed_info->info->info->version);
 					}
+					break;
+				default:
+					R_LOG_ERROR ("Unsupported rebase info version %d", trimmed_info->info->info->version);
+				}
+				if (pj) {
 					pj_end (pj);
+				} else if (sb) {
+					r_strbuf_append (sb, "\n");
 				}
 			}
-			r_list_free (infos);
 		}
+		r_list_free (infos);
+	}
 
-		r_list_free (slices);
+	r_list_free (slices);
 
+	if (pj) {
 		pj_end (pj);
+	}
 
+	if (pj) {
 		return pj_drain (pj);
 	}
+	if (sb) {
+		return r_strbuf_drain (sb);
+	}
+	return NULL;
+}
+
+static char *__system(RIO *io, RIODesc *fd, const char *command) {
+	r_return_val_if_fail (io && fd && fd->data && command, NULL);
+	RIODscObject *dsc = (RIODscObject*) fd->data;
+
+	if (r_str_startswith (command, "iP")) {
+		ut64 size = 8;
+		switch (command[2]) {
+		case '?':
+			io->cb_printf ("Usage: :iP[j?] [size]\n");
+			io->cb_printf (" :iP?   get this help message\n");
+			io->cb_printf (" :iP    show pointer metadata\n");
+			io->cb_printf (" :iPj   show pointer metadata in json\n\n");
+			return NULL;
+		case 'j':
+			if (command[3] == ' ') {
+				size = r_num_math (NULL, command + 4);
+			}
+			return __infoPointer (dsc, size, R_MODE_JSON);
+		case ' ':
+			size = r_num_math (NULL, command + 3);
+		case '\0':
+			return __infoPointer (dsc, size, R_MODE_PRINT);
+		}
+	} else if (command && command[0] == '?'){
```

Comment:
```suggestion
	} else if (command && command[0] == '?') {
```

Reply:
yeah i always forget to run `sys/lint.sh` locally first 😓